src/sentry/utils/request_cache.py

```diff
@@ -0,0 +1,33 @@
+import threading
+
+from django.core.signals import request_finished
+
+from sentry.models import OrganizationMember
+
+_cache = threading.local()
+
+
+def request_cache(func):
+    def wrapped(*args, **kwargs):
+        if not hasattr(_cache, "items"):
+            _cache.items = {}
+        cache_key = (func, repr(args), repr(kwargs))
```

Comment:
This will work ok as long as args/kwargs are not objects. For example, if one is a `Project`:
`"<Project at 0x114346f60: id=1, team_id=None, name='Internal', slug='internal'>"`

So this would be specific to that specific instance of  project 1, rather than any instances of project 1.

Not sure I have a great solution, but I think this sort of thing might be worth calling out in the comment for the decorator.

Reply:
@wedamija I think that's acceptable since it would lead to extra queries instead of returning the wrong value. I'll add a comment, good call.